### DISCUSSION TITLE
How to verify that it is GITHUB_TOKEN?

### DISCUSSION THREAD

**Discussion by yi-Xu-0100** [community/community#25411] [Actions]
<p>I verify whether it is <code>secret.GITHUB_TOKEN</code> by using <code>my_token === process.env.ACTIONS_RUNTIME_TOKEN</code>.</p>
<p>But it returns false, How can I verify the token is GITHUB_TOKEN ?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/yi-xu-0100">@yi-Xu-0100</a>,</p>
<p>There is no any <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables" rel="noopener nofollow ugc">default environment variable</a> named “<strong>ACTIONS_RUNTIME_TOKEN</strong>” for the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow" rel="noopener nofollow ugc"><strong>GITHUB_TOKEN</strong></a> secret.<br>
If you want to access the <strong>GITHUB_TOKEN</strong> secret via an environment variable, you need to map it as an environment variable in the workflow.<br>
For example:</p>
<pre><code class="lang-yaml">env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
</code></pre>
<br>
<p>The <strong>GITHUB_TOKEN</strong> secret has the following usage limitation:</p>
<ul>
<li>
<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#about-the-github_token-secret" rel="noopener nofollow ugc">The permissions of the <strong>GITHUB_TOKEN</strong> are limited to the repository that contains your workflow.</a></p>
</li>
<li>
<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow" rel="noopener nofollow ugc">When you use the repository’s <strong>GITHUB_TOKEN</strong> to perform tasks on behalf of the GitHub Actions app, events triggered by the <strong>GITHUB_TOKEN</strong> will not create a new workflow run.</a></p>
</li>
</ul>
<p>To verify whether a token is the <strong>GITHUB_TOKEN</strong>, you can use above points.<br>
For example, use the token in a workflow to trigger another workflow, such as push commits, execute the API “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#create-a-workflow-dispatch-event" rel="noopener nofollow ugc"><strong>Create a workflow dispatch event</strong></a>” or “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-repository-dispatch-event" rel="noopener nofollow ugc"><strong>Create a repository dispatch event</strong></a>”, etc… If the operation completed successfully but the workflow is not triggered, the token is the <strong>GITHUB_TOKEN</strong>.</p>

**Comment by yi-Xu-0100** [comment]
<p><a class="mention" href="/brightran">@brightran</a>,<br>
:heart:Thanks for your reply! But I just do not want to make any trigger for operation to verify the token. I want to check the permission of the token, just like the <a href="https://github.community/t/can-i-check-the-permissions-of-a-token/139486">Can I check the permissions of a token</a>. Is there any solution for this question? :thinking:</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/yi-xu-0100">@yi-Xu-0100</a>,</p>
<p>As I mentioned in that topic, we seem have no available methods to list all the access scopes of a token.</p><aside class="quote" data-post="2" data-topic="139486">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/can-i-check-the-permissions-of-a-token/139486/2">Can I check the permissions of a token</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    <a class="mention" href="/tombrus">@tombrus</a>, 
We seem have no available methods to list all the access scopes of a token. 
I tried using the <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/oauth-authorizations" rel="noopener nofollow ugc">OAuth Authorizations API</a>, but did not work. 
If you are the authenticated user of the token, you can try to navigate to Personal settings &gt; Developer settings &gt; Personal access tokens to view the access scopes of the tokens you have created.
  </blockquote>
</aside>
